src/assets/src/components/preferences.tsx

```diff
@@ -0,0 +1,168 @@
+import { redirectToLogin } from "../utils";
+import { PageProps } from "./page";
+
+const validatePhoneNumber = (phone: string, countryDialCode: string): Error[] => {
+    return [
+        (countryDialCode === '1' && phone.length !== 11)
+            && new Error("The phone number entered was invalid; USA phone numbers must have 11 digits."),
+    ].filter(e => e) as Error[];
+}
+
+interface PreferencesEditorProps {
+    user: MyUser;
+    disabled: boolean;
+    onUpdateInfo: (phoneNumber: string, notifyMeAttendee: boolean, notifyMeHost: boolean) => void;
+    errorOccurred: boolean;
+}
+
+type ValidationStatus = null | Error[]; // null = no changes, [] = valid
+
+function PreferencesEditor(props: PreferencesEditorProps) {
+    const [phoneField, setPhoneField] = useState(props.user.phone_number);
+    const [countryDialCode, setCountryDialCode] = useState("");
+    const [notifyMeAttendee, setNotifyMeAttendee] = useState(props.user.notify_me_attendee)
+    const [notifyMeHost, setNotifyMeHost] = useState(props.user.notify_me_host)
+    const [validationStatus, setValidationStatus] = useState(undefined as undefined | ValidationStatus);
+
+    const phoneInput = (
+        <PhoneInput
+            country={'us'}
+            value={props.user.phone_number}
+            onChange={(value: any, data: any) => {
+                setPhoneField(value);
+                if ('dialCode' in data) setCountryDialCode(data.dialCode);
+            }}
+            disabled={props.disabled}
+            inputProps={{id: 'phone'}}
+        />
+    );
+    const notifyMeAttendeeInput = (
+        <Form.Check 
+            type="checkbox"
+            id="notify-me-attendee"
+            disabled={!phoneField} 
+            checked={notifyMeAttendee} 
+            onChange={() => setNotifyMeAttendee(!notifyMeAttendee)}
+            label="As an attendee, notify me via SMS when it becomes my turn." />
+    );
+    const notifyMeHostInput = (
+        <Form.Check 
+            type="checkbox"
+            id="notify-me-host"
+            disabled={!phoneField} 
+            checked={notifyMeHost}
+            onChange={() => setNotifyMeHost(!notifyMeHost)}
+            label="As a host, notify me via SMS when someone joins my empty queue." />
+    );
+
+    const validateAndSubmit = (e: React.SyntheticEvent) => {
+        e.preventDefault() // Prevent page reload
+        const changed = props.user.phone_number !== phoneField
+            || props.user.notify_me_attendee !== notifyMeAttendee
+            || props.user.notify_me_host !== notifyMeHost;
+        if (!changed) {
+            setValidationStatus(null);
+            return;
+        }
+        if (phoneField.length <= 1) {
+            // Update phone number to be empty if they try to delete everything in the phone field
+            // Seems to be a known issue where the last character can't be removed as part of onChange:
+            // https://github.com/bl00mber/react-phone-input-2/issues/231
```

Comment:
According to this issue, it should be fixed and the fix should be in the latest release, but I was still getting the bug even after updating the package, doing a `docker-compose build`, etc. I might just need to clear volumes (`docker-compose down -v`), so I'll give that a shot before I give up.

Reply:
That did the trick, but the old behavior actually made sense, since a user could pick a `countryDialCode` (which autofills "1" etc.) and then not enter their phone number, and I think that state should be treated as "empty phoneField" for the purposes of this block. So I made some changes to submit without validating the phone number if they ONLY entered a `countryDialCode`, and to also leave the opt-ins disabled in that case until they enter more digits. I updated the library anyway and removed the comment.

Reply:
Thanks! I think you're saying the code I wrote before made sense, not the prior library implementation? Anyways, I think you have is working well and it's good to update the lib and remove the comment.

Reply:
Yeah, the library implementation did have a bug, but checking `phoneField <= 1` still made sense (to me at least) for users with a 1-digit `dialCode`.